File: LSP/Socket.java
public class Socket {
    public boolean powerIsOn;

    public Socket() {
        powerIsOn = true;
    }

    public void powerOn() {
        powerIsOn = true;
        System.out.println("Power is ON");
    }

    public void powerOff() {
        powerIsOn = false;
        System.out.println("Power is OFF");
    }
}

File: LSP/LSPDemo/Socket.cs
using System;

namespace MMBakker.SolidDesignPrinciples.LiskovSubstitution
{
    public class Socket
    {
        public bool PowerIsOn { get; set; }
        public Socket()
        {
            PowerIsOn = true;
        }
        public void PowerOn()
        {
            PowerIsOn = true;
            Console.WriteLine("Power is ON");
        }

        public void PowerOff()
        {
            PowerIsOn = false;
            Console.WriteLine("Power is OFF");
        }
    }
}
